### ISSUE TITLE
Allow setting multiple cookies

### ISSUE THREAD

**Issue by hugo** [remix-run/remix#92]
**tl;dr** Remix currently doesn't allow setting multiple cookies at once. This is in line with the underlying spec of the Response objects Remix deals with, but is kinda annoying IRL.

---

Cookies are set in `loaders` and `actions` using the `Set-Cookie` header. Remix provides a number of ways to set headers in responses. You can (that I know of) return plain key/value object, an array of key/value pairs, or a [Headers](https://developer.mozilla.org/en-US/docs/Web/API/Headers) object. These all eventually end up in the same place. Which , unfortunately for us, is the problem. None of these methods, at preset, allows setting multiple `Set-Cookie` headers.

Because of it origins in the browser, the Headers object has a particularly strict way of handling headers. To whit, it attempts to fold headers together. The `#append` method does this by joining multiple values for the same header with comma separation. This is according to the spec and matches how servers, proxies, etc. should work as well. With one exception. `Set-Cookie` needs to be special cased as a comma has semantic meaning within the content of the header value and so cannot safely be used as a separator. To quote the most recent [relevant RFC](https://tools.ietf.org/html/rfc7230#section-3.2.2):
>Note: In practice, the "Set-Cookie" header field ([RFC6265]) often appears multiple times in a response message and does not use the list syntax, violating the above requirements on multiple header fields with the same name.  Since it cannot be combined into a single field-value, recipients ought to handle "Set-Cookie" as a special case while processing header fields.

This isn't a problem in the browser, because the `fetch` API has a special case, also. The `fetch` client is not allowed to specify a `Set-Cookie` header. This makes sense, how would you set a cookie in a server?

I hacked a very quick and dirty fix for this by editing `@remix-run/express/index.js`:
```js
sendPlatformResponse(remixResponse, _req, res) {
  res.status(remixResponse.status);

  for (let [key, value] of remixResponse.headers.entries()) {
    // here begineth the hack
    if (key === "set-cookie") {
      res.set(key, value.split(","))
    } else {
      res.set(key, value);
    }
    // here endeth the hack
  }

  if (Buffer.isBuffer(remixResponse.body)) {
    res.end(remixResponse.body);
  } else {
    remixResponse.body.pipe(res);
  }
}
```
This isn't a production ready fix because it will break in the case that `Expires` is set on a cookie. I got around this by specifying `MaxAge` instead.

---

I'll be honest, I don't know the best way for Remix to handle this.  Building on top of the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) is pretty foundational to Remix's design. Given that this is a genuine special case, maybe it's okay for Remix to deviate from the spec in this one area?
